app/models/report_form.rb

```diff
@@ -7,17 +8,47 @@ class ReportForm
     attr_accessor "development_#{n}"
   end
 
+  (1..SMART_OBJECTIVES).each do |n|
+    attr_accessor "smart_what_#{n}"
+    attr_accessor "smart_how_#{n}"
+  end
+
   def development_as_json
     (1..DEVELOPMENT_OBJECTIVES).map { |n| send("development_#{n}") || '' }
   end
 
+  def smart_as_json
+    (1..SMART_OBJECTIVES).map do |n|
+      {
+        what: send("smart_what_#{n}") || '',
+        how: send("smart_how_#{n}") || ''
+      }
+    end
+  end
+
   def self.from_report(report)
```

Comment:
This is the factory method I was talking about. I really don't think I should extract it to a separate class, but because of the complexity I had to split it and now there are 2 more public methods which make no sense to be public. If that makes sense..